File: template.rb
#!/usr/bin/ruby

IS_DEBUG_ENABLED = true;
IS_READ_FROM_FILE = true;
DATA_FILE = "template.txt";

def read_lines_from_file()
  File.open(DATA_FILE) do |file|
    return file.read.split("\n")
  end
end

def read_lines_from_stdin()
  return $stdin.read.split("\n").select { |line| line.length > 0 }
end

def lines_to_list_of_str_list(lines)
  return lines.map { |line| line.split(" ") }
end

def lines_to_list_of_int_list(lines)
  return lines.map { |line|
    line.split(" ").map { |e2| e2.to_i }
  }
end

def debug(msg, *args)
  if IS_DEBUG_ENABLED then
    printf(msg+"\n", *args)
  end
end

def dump(obj)
  if IS_DEBUG_ENABLED then
    p (obj)
  end
end

def main()
  if IS_READ_FROM_FILE then
    input_lines = read_lines_from_file()
  else
    input_lines = read_lines_from_stdin()
  end

  debug("input_lines=")
  dump(input_lines)

  input_values = lines_to_list_of_int_list(input_lines)
  debug("input_values=")
  dump(input_values)

end

main()

File: template.go
package main

import (
  "bufio"
  "fmt"
  "os"
  "strconv"
  "strings"
)

var isDebugEnabled = true
var isReadFromFile = true
var dataFile = "template.txt"

func debug(formatString string, args ...interface{}) {
  if ! isDebugEnabled {
    return
  }

  fmt.Printf(formatString + "\n", args...)
}

func readLinesFromFile() []string {
  f, err := os.Open(dataFile);

  if err != nil {
    fmt.Fprintf(os.Stderr, "could not read file: %v\n", dataFile, err)
    os.Exit(1)
  }

  defer f.Close()

  scanner := bufio.NewScanner(f)
  lines := []string{}
  for scanner.Scan() {
    line := scanner.Text()
    if (len(line) > 0) {
      lines = append(lines, line)
    }
  }

  return lines;
}

func readLinesFromStdin() []string {
  scanner := bufio.NewScanner(os.Stdin)
  lines := []string{}
  for scanner.Scan() {
    line := scanner.Text()
    if (len(line) > 0) {
      lines = append(lines, line)
    }
  }

  return lines;
}

func linesToStrList(lines []string) [][]string {
  var values [][]string

  for _, line := range(lines) {
    values = append(values, strings.Split(line, " "))
  }

  return values
}

func linesToIntList(lines []string) [][]int {
  var values [][]int

  for _, line := range(lines) {
    strValues := strings.Split(line, " ")
    var intValues []int

    for _, v := range(strValues) {
      intValue, _ := strconv.Atoi(v)
      intValues = append(intValues, intValue)
    }

    values = append(values, intValues)
  }

  return values
}

func main() {
  var inputLines []string

  if isReadFromFile {
    inputLines = readLinesFromFile()
  } else {
    inputLines = readLinesFromStdin()
  }

  debug("inputLines=[%s]\n", strings.Join(inputLines, ","))

  inputValues := linesToIntList(inputLines)
  debug("inputValues=%s\n", inputValues)
}
